auslib/db.py

```diff
@@ -1040,11 +1060,33 @@ def addRelease(self, name, product, blob, changed_by, transaction=None):
     def updateRelease(self, name, changed_by, old_data_version, product=None, read_only=None, blob=None, transaction=None):
         if product or blob:
             self._proceedIfNotReadOnly(name, transaction=transaction)
+        where = [self.name == name]
         what = {}
-        if read_only is not None:
-            what['read_only'] = read_only
+
+        current_product = self.select(where=where, columns=[self.product], transaction=transaction)[0]["product"]
```

Comment:
I probably should have started this comment a couple of lines lower, but it looks to me that we require release:modify (on the current product) for all changes, as well as release_read_only if that's the change being made. It was mostly a comment in passing, to see if that's what you intended. Needn't block.